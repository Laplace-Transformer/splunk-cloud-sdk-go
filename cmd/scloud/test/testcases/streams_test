streams compile --input-datafile testcases/test_stream_compile.spl

streams create-pipeline --name test-passthrough --input-datafile testcases/test_stream_create_pipeline.json --bypass-validation true --description "A passthrough pipeline"
streams validate-pipeline --input-datafile testcases/test_stream_validate_pipeline.json
streams update-pipeline  --id 57cc9796-bb08-48dd-9f0e-f269fc762eda  --input-datafile testcases/test_stream_patch_pipeline.json --description "Updated Integration Test Pipeline testPipelinfe1582068339" --name "updatedtestPipelinfe1582068339"
streams patch-pipeline --id 57cc9796-bb08-48dd-9f0e-f269fc762eda --name passthrough --input-datafile testcases/test_stream_patch_pipeline.json --bypass-validation true --description "Patched pipeline"

streams reactivate-pipeline  --id 57cc9796-bb08-48dd-9f0e-f269fc762eda

streams list-pipelines
streams get-pipeline --id 57cc9796-bb08-48dd-9f0e-f269fc762eda
streams get-pipelines-status
streams activate-pipeline --id 57cc9796-bb08-48dd-9f0e-f269fc762eda
streams deactivate-pipeline --id 57cc9796-bb08-48dd-9f0e-f269fc762eda
streams delete-pipeline --id 280afcdd-cca1-4cff-b643-9d3295a70075

streams list-connectors
streams list-connections

streams create-connection --connector-id 879837b0-cabf-4bc2-8589-fcc4dad753e7 --data "{\"splunk-url\":\"https://hostname.port\", \"token\":\"mytoken\"}" --description "newconnection" --name "myconnection"
streams update-connection --connection-id 879837b0-cabf-4bc2-8589-fcc4dad753e7 --data "{\"splunk-url\":\"https://hostname.port\", \"token\":\"mytoken\"}" --description "newconnection" --name "myconnection"
streams put-connection --connection-id 879837b0-cabf-4bc2-8589-fcc4dad753e7 --data "{\"splunk-url\":\"https://hostname.port\", \"token\":\"mytoken\"}" --description "newconnection" --name "myconnection"

streams delete-connection --connection-id 6c7896b2-a53c-4078-b757-07e80739d91e

streams get-registry --local false

streams create-template --input-datafile testcases/test_stream_create_template.json --name testTemplate1582068227 --description "integration test template"
streams list-templates
streams get-template --template-id 680db4a4-d733-428a-a0a0-9eac9f3e8252

#Deprecated in v3beta1
streams create-group --input-datafile testcases/test_stream_create_group.json

streams get-input-schema  --node-uuid "3682041e-1814-4bd7-8e60-5c889b822585" --target-port-name "input" --input-datafile testcases/test_stream_get_input_schema.json

# go sdk dosesn't have a working example for GetOutputSchema endpoint
#streams get-output-schema --node-uuid "3682041e-1814-4bd7-8e60-5c889b822585" --target-port-name "output" --input-datafile testcases/test_stream_get_output_schema.json

