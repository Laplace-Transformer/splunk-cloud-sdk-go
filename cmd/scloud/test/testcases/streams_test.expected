#testcase: streams compile --input-datafile testcases/test_stream_compile.spl
REQUEST URL:streams/v3beta1/pipelines/compile
REQUEST BODY:{{"spl":"| from splunk_firehose() | into index(\"index\", \"main\");"}}

#testcase: streams create-pipeline --name test-passthrough --input-datafile testcases/test_stream_create_pipeline.json --bypass-validation true --description "A passthrough pipeline"
REQUEST URL:streams/v3beta1/pipelines
REQUEST BODY:{{"data":{"edges":[{"sourceNode":"c0edd974-1190-3f73-923a-4330a5bc2a7c","sourcePort":"output","targetNode":"6defeeb3-a0d1-386f-897b-7cff5fe2ce34","targetPort":"input"}],"nodes":[{"id":"c0edd974-1190-3f73-923a-4330a5bc2a7c","op":"from_splunk_firehose","resolvedId":"from_splunk_firehose"},{"id":"6defeeb3-a0d1-386f-897b-7cff5fe2ce34","op":"into_index","arguments":{"dataset":"\"main\"","module":"\"index\""},"resolvedId":"into_index:collection\u003crecord\u003cR\u003e\u003e:expression\u003cstring\u003e:expression\u003cstring\u003e"}]},"name":"test-passthrough","bypassValidation":true,"description":"A passthrough pipeline"}}

#testcase: streams validate-pipeline --input-datafile testcases/test_stream_validate_pipeline.json
REQUEST URL:streams/v3beta1/pipelines/validate
REQUEST BODY:{{"upl":{"edges":[{"sourceNode":"c0edd974-1190-3f73-923a-4330a5bc2a7c","sourcePort":"output","targetNode":"6defeeb3-a0d1-386f-897b-7cff5fe2ce34","targetPort":"input"}],"nodes":[{"id":"c0edd974-1190-3f73-923a-4330a5bc2a7c","op":"from_splunk_firehose","resolvedId":"from_splunk_firehose"},{"id":"6defeeb3-a0d1-386f-897b-7cff5fe2ce34","op":"into_index","arguments":{"dataset":"\"main\"","module":"\"index\""},"resolvedId":"into_index:collection\u003crecord\u003cR\u003e\u003e:expression\u003cstring\u003e:expression\u003cstring\u003e"}]}}}

#testcase: streams update-pipeline  --id 57cc9796-bb08-48dd-9f0e-f269fc762eda  --input-datafile testcases/test_stream_patch_pipeline.json --description "Updated Integration Test Pipeline testPipelinfe1582068339" --name "updatedtestPipelinfe1582068339"
REQUEST URL:streams/v3beta1/pipelines/57cc9796-bb08-48dd-9f0e-f269fc762eda
REQUEST BODY:{{"data":{"edges":[{"sourceNode":"c0edd974-1190-3f73-923a-4330a5bc2a7c","sourcePort":"output","targetNode":"6defeeb3-a0d1-386f-897b-7cff5fe2ce34","targetPort":"input"}],"nodes":[{"id":"c0edd974-1190-3f73-923a-4330a5bc2a7c","op":"from_splunk_firehose","resolvedId":"from_splunk_firehose"},{"id":"6defeeb3-a0d1-386f-897b-7cff5fe2ce34","op":"into_index","arguments":{"dataset":"\"main\"","module":"\"index\""},"resolvedId":"into_index:collection\u003crecord\u003cR\u003e\u003e:expression\u003cstring\u003e:expression\u003cstring\u003e"}]},"name":"updatedtestPipelinfe1582068339","description":"Updated Integration Test Pipeline testPipelinfe1582068339"}}

#testcase: streams patch-pipeline --id 57cc9796-bb08-48dd-9f0e-f269fc762eda --name passthrough --input-datafile testcases/test_stream_patch_pipeline.json --bypass-validation true --description "Patched pipeline"
REQUEST URL:streams/v3beta1/pipelines/57cc9796-bb08-48dd-9f0e-f269fc762eda
REQUEST BODY:{{"bypassValidation":true,"data":{"edges":[{"sourceNode":"c0edd974-1190-3f73-923a-4330a5bc2a7c","sourcePort":"output","targetNode":"6defeeb3-a0d1-386f-897b-7cff5fe2ce34","targetPort":"input"}],"nodes":[{"id":"c0edd974-1190-3f73-923a-4330a5bc2a7c","op":"from_splunk_firehose","resolvedId":"from_splunk_firehose"},{"id":"6defeeb3-a0d1-386f-897b-7cff5fe2ce34","op":"into_index","arguments":{"dataset":"\"main\"","module":"\"index\""},"resolvedId":"into_index:collection\u003crecord\u003cR\u003e\u003e:expression\u003cstring\u003e:expression\u003cstring\u003e"}]},"description":"Patched pipeline","name":"passthrough"}}

#testcase: streams reactivate-pipeline  --id 57cc9796-bb08-48dd-9f0e-f269fc762eda
REQUEST URL:streams/v3beta1/pipelines/57cc9796-bb08-48dd-9f0e-f269fc762eda/reactivate
REQUEST BODY:{{}}

#testcase: streams list-pipelines
REQUEST URL:streams/v3beta1/pipelines
REQUEST BODY:<nil>

#testcase: streams get-pipeline --id 57cc9796-bb08-48dd-9f0e-f269fc762eda
REQUEST URL:streams/v3beta1/pipelines/57cc9796-bb08-48dd-9f0e-f269fc762eda
REQUEST BODY:<nil>

#testcase: streams get-pipelines-status
REQUEST URL:streams/v3beta1/pipelines/status
REQUEST BODY:<nil>

#testcase: streams activate-pipeline --id 57cc9796-bb08-48dd-9f0e-f269fc762eda
REQUEST URL:streams/v3beta1/pipelines/57cc9796-bb08-48dd-9f0e-f269fc762eda/activate
REQUEST BODY:{{}}

#testcase: streams deactivate-pipeline --id 57cc9796-bb08-48dd-9f0e-f269fc762eda
REQUEST URL:streams/v3beta1/pipelines/57cc9796-bb08-48dd-9f0e-f269fc762eda/deactivate
REQUEST BODY:{{}}

#testcase: streams delete-pipeline --id 280afcdd-cca1-4cff-b643-9d3295a70075
REQUEST URL:streams/v3beta1/pipelines/280afcdd-cca1-4cff-b643-9d3295a70075
REQUEST BODY:<nil>

#testcase: streams list-connectors
REQUEST URL:streams/v3beta1/connectors
REQUEST BODY:<nil>

#testcase: streams list-connections
REQUEST URL:streams/v3beta1/connections
REQUEST BODY:<nil>

#testcase: streams create-connection --connector-id 879837b0-cabf-4bc2-8589-fcc4dad753e7 --data "{\"splunk-url\":\"https://hostname.port\", \"token\":\"mytoken\"}" --description "newconnection" --name "myconnection"
REQUEST URL:streams/v3beta1/connections
REQUEST BODY:{{"connectorId":"879837b0-cabf-4bc2-8589-fcc4dad753e7","data":{"splunk-url":"https://hostname.port","token":"mytoken"},"description":"newconnection","name":"myconnection"}}

#testcase: streams update-connection --connection-id 879837b0-cabf-4bc2-8589-fcc4dad753e7 --data "{\"splunk-url\":\"https://hostname.port\", \"token\":\"mytoken\"}" --description "newconnection" --name "myconnection"
REQUEST URL:streams/v3beta1/connections/879837b0-cabf-4bc2-8589-fcc4dad753e7
REQUEST BODY:{{"data":{"splunk-url":"https://hostname.port","token":"mytoken"},"description":"newconnection","name":"myconnection"}}

#testcase: streams put-connection --connection-id 879837b0-cabf-4bc2-8589-fcc4dad753e7 --data "{\"splunk-url\":\"https://hostname.port\", \"token\":\"mytoken\"}" --description "newconnection" --name "myconnection"
REQUEST URL:streams/v3beta1/connections/879837b0-cabf-4bc2-8589-fcc4dad753e7
REQUEST BODY:{{"data":{"splunk-url":"https://hostname.port","token":"mytoken"},"description":"newconnection","name":"myconnection"}}

#testcase: streams delete-connection --connection-id 6c7896b2-a53c-4078-b757-07e80739d91e
REQUEST URL:streams/v3beta1/connections/6c7896b2-a53c-4078-b757-07e80739d91e
REQUEST BODY:<nil>

#testcase: streams get-registry --local false
REQUEST URL:streams/v3beta1/pipelines/registry?local=false
REQUEST BODY:<nil>

#testcase: streams create-template --input-datafile testcases/test_stream_create_template.json --name testTemplate1582068227 --description "integration test template"
REQUEST URL:streams/v3beta1/templates
REQUEST BODY:{{"data":{"edges":[{"sourceNode":"2c5aa0e6-ccc5-3577-9f12-5d8c466c4e84","sourcePort":"output","targetNode":"ba342517-84db-302e-8fb7-545b49752cc3","targetPort":"input"},{"sourceNode":"ba342517-84db-302e-8fb7-545b49752cc3","sourcePort":"output","targetNode":"8a9a491b-ed6d-3ced-8ad5-9eb966d3c226","targetPort":"input"},{"sourceNode":"8a9a491b-ed6d-3ced-8ad5-9eb966d3c226","sourcePort":"output","targetNode":"2c43f86e-ad69-3ce1-8fdb-8f1b59ba5370","targetPort":"input"},{"sourceNode":"2c43f86e-ad69-3ce1-8fdb-8f1b59ba5370","sourcePort":"output","targetNode":"3ab66bd5-c6f4-3192-9123-54467d7feabf","targetPort":"input"}],"nodes":[{"id":"2c5aa0e6-ccc5-3577-9f12-5d8c466c4e84","op":"from_generate_events","arguments":{"interval":1000,"type":"timestamp"},"attributes":{"user":{"name":"Generate Events: Generate Sample Data","notes":""}},"resolvedId":"from_generate_events:string:long"},{"id":"ba342517-84db-302e-8fb7-545b49752cc3","op":"eval","arguments":{"function":["body=concat(\"0XGFSod1vIoyZeTdldL2vR4wbWZFRTo25,qVbCl1Yn2XYhp7wQVzBE0ik7kI2pAsUI,Credit Card,1615.65,\", strftime(now()-5000, \"%Y-%m-%d %H:%M:%S\", null), \",\", strftime(now(), \"%Y-%m-%d %H:%M:%S\", null), \",-73.953948,40.771702,2475601523770875\")"]},"attributes":{"user":{"name":"Eval: Format Sample Events","notes":""}},"resolvedId":"eval:collection\u003crecord\u003cR\u003e\u003e:collection\u003cexpression\u003cany\u003e\u003e"},{"id":"8a9a491b-ed6d-3ced-8ad5-9eb966d3c226","op":"eval","arguments":{"function":["body=extract_regex(body, /(?\u003ctid\u003e[^,]+),(?\u003ccid\u003e[^,]+),(?\u003cType\u003e[^,]+),(?\u003cAmount\u003e[^,]+),(?\u003csdate\u003e\\S+)\\s(?\u003cstime\u003e\\S+),(?\u003cedate\u003e\\S+)\\s(?\u003cetime\u003e\\S+?),(?\u003cLongitude\u003e\\S+?),(?\u003cLatitude\u003e\\S+?),(?\u003cCard\u003e\\d*)/)","body=map_set(body, \"Card\", \"\u003credacted\u003e\")","source_type=\"cc_transactions:json\"","timestamp=strptime(concat(body.sdate, \" \", body.stime), \"%Y-%m-%d %H:%M:%S\", \"UTC\")"]},"attributes":{"user":{"name":"Eval: Redact Credit Card Numbers","notes":""}},"resolvedId":"eval:collection\u003crecord\u003cR\u003e\u003e:collection\u003cexpression\u003cany\u003e\u003e"},{"id":"2c43f86e-ad69-3ce1-8fdb-8f1b59ba5370","op":"fields","arguments":{"field_list":["timestamp","body","source_type"],"operator":"+"},"attributes":{"user":{"name":"Fields: Keep Specific Fields","notes":""}},"resolvedId":"fields:collection\u003crecord\u003cR\u003e\u003e:collection\u003cfield\u003e"},{"id":"3ab66bd5-c6f4-3192-9123-54467d7feabf","op":"into_dev_null","attributes":{"user":{"name":"Send to Null: Placeholder for Destination","notes":""}},"resolvedId":"into_dev_null:collection\u003crecord\u003cR\u003e\u003e"}],"attributes":{"activeNodeId":"2c5aa0e6-ccc5-3577-9f12-5d8c466c4e84"}},"description":"integration test template","name":"testTemplate1582068227"}}

#testcase: streams list-templates
REQUEST URL:streams/v3beta1/templates
REQUEST BODY:<nil>

#testcase: streams get-template --template-id 680db4a4-d733-428a-a0a0-9eac9f3e8252
REQUEST URL:streams/v3beta1/templates/680db4a4-d733-428a-a0a0-9eac9f3e8252
REQUEST BODY:<nil>

#testcase: streams create-group --input-datafile testcases/test_stream_create_group.json
unknown flag: --input-datafile

#testcase: streams get-input-schema  --node-uuid "3682041e-1814-4bd7-8e60-5c889b822585" --target-port-name "input" --input-datafile testcases/test_stream_get_input_schema.json
REQUEST URL:streams/v3beta1/pipelines/input-schema
REQUEST BODY:{{"nodeUuid":"3682041e-1814-4bd7-8e60-5c889b822585","targetPortName":"input","uplJson":{"edges":[{"sourceNode":"1df22bab-8d0d-44c3-ad8a-f3ba1b85ff84","sourcePort":"output","targetNode":"3682041e-1814-4bd7-8e60-5c889b822585","targetPort":"input"}],"nodes":[{"id":"1df22bab-8d0d-44c3-ad8a-f3ba1b85ff84","op":"read-splunk-firehose","resolvedId":"read-splunk-firehose"},{"id":"3682041e-1814-4bd7-8e60-5c889b822585","op":"write-index","resolvedId":"write-index:collection\u003crecord\u003cR\u003e\u003e:expression\u003cstring\u003e:expression\u003cstring\u003e"}]}}}

